shared/signup/phone-number/verify.tsx

```diff
@@ -3,105 +3,101 @@ import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
 import {SignupScreen} from '../common'
 
-type Props = {
+export type Props = {
   error: string
   onBack: () => void
-  onChangeCode: (code: string) => void
-  onContinue: () => void
+  onCancel: () => void
+  onContinue: (code: string) => void
   onResend: () => void
   phoneNumber: string
   resendWaiting: boolean
+  verifyWaiting: boolean
 }
 
-class VerifyPhoneNumber extends React.Component<
-  Props,
-  {
-    value: string
-  }
-> {
-  state = {value: ''}
-  _onChangeText = value => {
-    this.props.onChangeCode(value)
-    this.setState({value})
-  }
-  render() {
-    return (
-      <SignupScreen
-        onBack={this.props.onBack}
-        banners={this.props.error ? [<Kb.Banner key="error" color="red" text={this.props.error} />] : []}
-        buttons={[{label: 'Continue', onClick: this.props.onContinue, type: 'Success'}]}
-        titleComponent={
+const VerifyPhoneNumber = (props: Props) => {
+  // Clear data on unmount
+  React.useEffect(() => {
+    return props.onCancel
```

Comment:
did you mean to call `onCancel` instead of returning it? I don't understand what this is doing if not.

Reply:
No, this is `componentWillUnmount` a-la hooks. We return the callback that should be called on unmount and updates. I actually think this is a pretty bad pattern, it is also called when props in the second array change, so I imagine it could lead to terrible unexpected behavior if something in the connector changes.